common/dictionary-db/dictionary-db.ts

```diff
@@ -0,0 +1,1416 @@
+
+            noteInfo.mod = modifiedAt;
+            modifiedNotes.push(noteInfo);
+            for (const cardId of noteInfo.cards) modifiedCardIdsSet.add(cardId);
+        }
+
+        const modifiedCardsDeck: Map<number, string> = new Map();
+        if (modifiedNotes.length) {
+            const modifiedCardIds = Array.from(modifiedCardIdsSet);
+            for (const cardInfo of await anki.cardsInfo(modifiedCardIds)) {
+                modifiedCardsDeck.set(cardInfo.cardId, cardInfo.deckName); // cardsInfo is much slower than notesInfo so we try to call it only if needed
+            }
+            const dts = Array.from(trackStates.values()).map((ts) => ts.dt);
+            for (let i = modifiedNotes.length - 1; i >= 0; i--) {
+                let modified = false;
+                for (const dt of dts) {
+                    if (!modifiedNotes[i].cards.some((c) => hasDeck(dt, modifiedCardsDeck.get(c)!))) continue;
+                    if (!hasField(dt, Object.keys(modifiedNotes[i].fields))) continue;
+                    modified = true;
+                    break;
+                }
+                if (!modified) {
+                    for (const cardId of modifiedNotes[i].cards) modifiedCardIdsSet.delete(cardId);
+                    modifiedNotes.splice(i, 1);
+                }
+            }
+        }
+
+        if (modifiedNotes.length) {
+            const modifiedCardIds = Array.from(modifiedCardIdsSet);
+            const suspendedCards = new Set<number>();
+            const areSuspended = await anki.areSuspended(modifiedCardIds);
+            for (let i = 0; i < modifiedCardIds.length; i++) {
+                if (areSuspended[i]) suspendedCards.add(modifiedCardIds[i]);
+            }
+
+            for (const modifiedNote of modifiedNotes) {
+                const fields = new Map<string, string>();
+                for (const [fieldName, { value }] of Object.entries(modifiedNote.fields)) {
+                    const trimmedValue = value.trim();
+                    if (!trimmedValue.length) continue;
+                    fields.set(fieldName, trimmedValue);
+                }
+                for (const cardId of modifiedNote.cards) {
+                    modifiedCards.set(cardId, {
+                        noteId: modifiedNote.noteId,
+                        deckName: modifiedCardsDeck.get(cardId)!,
+                        fields,
+                        modifiedAt: modifiedNote.mod,
+                        statuses: new Map(),
+                        suspended: suspendedCards.has(cardId),
+                    });
+                }
+            }
+
+            for (const existingAnkiCards of existingAnkiNoteIdMap.values()) {
+                for (const ankiCard of existingAnkiCards) {
+                    if (!modifiedCardIdsSet.has(ankiCard.cardId)) continue;
+                    if (suspendedCards.has(ankiCard.cardId)) {
+                        if (!ankiCard.suspended) cardIdsToSuspend.push(ankiCard.cardId);
```

Comment:
I guess this may be due to a quirk of Anki Connect's API but why does suspended state need to be dealt with separately? I see your comment above that `Reviews or suspension status are reflected in card mod time`, so why isn't suspension status already taken care of by `modifiedCards`?